mldsa/sign.c

```diff
@@ -155,19 +155,140 @@ __contract__(
   shake256_squeeze(out, outlen, &state);
 }
 
+#define NONCE_UB ((UINT16_MAX - MLDSA_L) / MLDSA_L)
+
+/*************************************************
+ * Name:        attempt_signature_generation
+ *
+ * Description: Attempts to generate a single signature.
+ *              This function encapsulates one iteration of the
+ *              rejection sampling process to help CBMC analysis.
+ *
+ * Arguments:   - uint8_t *sig: pointer to output signature
+ *              - const uint8_t *mu: pointer to message hash
+ *              - const uint8_t *rhoprime: pointer to randomness seed
+ *              - uint16_t nonce: current nonce value
+ *              - const polyvecl mat[MLDSA_K]: expanded matrix
+ *              - const polyvecl *s1: secret vector s1
+ *              - const polyveck *s2: secret vector s2
+ *              - const polyveck *t0: secret vector t0
+ *
+ * Returns 0 if signature generation succeeds, -1 if rejected
```

Comment:
Please add a ` Reference:` comment here on how this differs from the reference implementation.
I remember you did make a couple of other changes as well. Maybe now is a good time to write them down if you still remember them.
That will save us some time further down the line.

Reply:
Done. Also added Reference: comment in sign.h for crypto_sign_signature_internal() to mention it can return -1 in the case of nonce exhaustion.